packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts

```diff
@@ -1215,5 +1222,24 @@ const a: Default = '';
         },
       ],
     },
+    {
```

Comment:
another case to think about

```ts
import type A from 'foo';
import B from 'foo';
let foo: A;
let bar = B;
```

What would this option do here?
I would expect that it would error

Reply:
This is the same case as described above and in this case I would not expect the rule to fail for the reasons described above.
There would of course still be the option of to change the configuration from `{ prefer: 'type-imports-mixed' }` to something like `{ prefer: 'type-imports', allowMixed: true }` and `{ prefer: 'no-type-imports', allowMixed: true }`